test/caption-stream.test.js

```diff
@@ -1536,6 +1541,112 @@ QUnit.test('the roll-up count can be changed on-the-fly', function() {
   QUnit.equal(captions.length, 0, 'cleared the caption');
 });
 
+QUnit.test('switching to roll-up from pop-on wipes memories and flushes captions', function() {
+  var captions = [];
+  cea608Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  [
+    { pts: 0 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 0 * 1000, ccData: characters('hi'), type: 0 },
+    // flip memories
+    { pts: 1 * 1000, ccData: 0x142f, type: 0 },
+    { pts: 1 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 2 * 1000, ccData: characters('oh'), type: 0 },
+    { pts: 2 * 1000, ccData: 0x142f, type: 0 },
+    { pts: 3 * 1000, ccData: 0x1425, type: 0 }
+  ].forEach(cea608Stream.push, cea608Stream);
+
+  var displayed = cea608Stream.displayed_.reduce(function(acc, val) {
+    acc += val;
+    return acc;
+  });
+  var nonDisplayed = cea608Stream.nonDisplayed_.reduce(function(acc, val) {
+    acc += val;
+    return acc;
+  });
+
+  QUnit.equal(captions.length, 2, 'both captions flushed');
+  QUnit.equal(displayed, '', 'displayed memory is wiped');
+  QUnit.equal(nonDisplayed, '', 'non-displayed memory is wiped');
+  QUnit.deepEqual(captions[0], {
+    startPts: 1000,
+    endPts: 2000,
+    text: 'hi',
+    stream: 'CC1'
+  }, 'first caption correct');
+  QUnit.deepEqual(captions[1], {
+    startPts: 2000,
+    endPts: 3000,
+    text: 'oh',
+    stream: 'CC1'
+  }, 'second caption correct');
+});
+
+QUnit.test('switching to roll-up from paint-on wipes memories and flushes captions', function() {
+  var captions = [];
+  cea608Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  [
+    { pts: 0 * 1000, ccData: 0x1429, type: 0 },
+    { pts: 0 * 1000, ccData: characters('hi'), type: 0 },
+    { pts: 1 * 1000, ccData: 0x1425, type: 0 }
+  ].forEach(cea608Stream.push, cea608Stream);
+
+  var displayed = cea608Stream.displayed_.reduce(function(acc, val) {
+    acc += val;
+    return acc;
+  });
+  var nonDisplayed = cea608Stream.nonDisplayed_.reduce(function(acc, val) {
+    acc += val;
+    return acc;
+  });
+
+  QUnit.equal(captions.length, 1, 'flushed caption');
+  QUnit.equal(displayed, '', 'displayed memory is wiped');
+  QUnit.equal(nonDisplayed, '', 'non-displayed memory is wiped');
+  QUnit.deepEqual(captions[0], {
+    startPts: 0,
+    endPts: 1000,
+    text: 'hi',
+    stream: 'CC1'
+  }, 'caption correct');
+});
+
+// NOTE: This should change to not wiping the display when caption
+// positioning is properly implemented
+QUnit.test('switching to paint-on from pop-on flushes display', function() {
+  var captions = [];
+  cea608Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  [
+    { pts: 0 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 0 * 1000, ccData: 0x1450, type: 0 },
+    { pts: 0 * 1000, ccData: characters('hi'), type: 0 },
+    { pts: 1 * 1000, ccData: 0x142f, type: 0 },
+    { pts: 1 * 1000, ccData: 0x1420, type: 0 },
+    // flip memories
```

Comment:
I think this comment is in the wrong place? Also, just adding descriptions to the packets would be helpful 😄